From 661f6f15d9d645acd748c7b8eac510099f124cab Mon Sep 17 00:00:00 2001
From: Leon Leijssen <leon.leijssen@topic.nl>
Date: Mon, 4 May 2020 05:03:47 -0700
Subject: [PATCH 1362/1852] drm: xlnx: pl_disp: fix missing
 xlnx_crtc_unregister

commit e466ff332e2dd1513f53933c5efd374c0337b69d from
https://github.com/Xilinx/linux-xlnx.git

During shutdown of the system WARN_ON(!list_empty(&helper->xlnx_crtcs)) in
xlnx_crtc.c xlnx_crtc_helper_fini() was being triggered. Fix it by
unregistering the registered crtc.

Signed-off-by: Leon Leijssen <leon.leijssen@topic.nl>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/xlnx_pl_disp.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/drm/xlnx/xlnx_pl_disp.c b/drivers/gpu/drm/xlnx/xlnx_pl_disp.c
index a1d720d0271a..f5f1ce64266d 100644
--- a/drivers/gpu/drm/xlnx/xlnx_pl_disp.c
+++ b/drivers/gpu/drm/xlnx/xlnx_pl_disp.c
@@ -524,6 +524,7 @@ static void xlnx_pl_disp_unbind(struct device *dev, struct device *master,
 {
 	struct xlnx_pl_disp *xlnx_pl_disp = dev_get_drvdata(dev);
 
+	xlnx_crtc_unregister(xlnx_pl_disp->drm, &xlnx_pl_disp->xlnx_crtc);
 	drm_plane_cleanup(&xlnx_pl_disp->plane);
 	drm_crtc_cleanup(&xlnx_pl_disp->xlnx_crtc.crtc);
 }
-- 
2.31.1

